### PR TITLE
Add version mismatch warnings

### PR DESCRIPTION
**Pull request by zetlen**
## Description
Adds version mismatch detection and logging.

- Modifies the bundler script to embed the version number in `package.json` as a constant.
- Compares `version` strings in the handshake messages used to connect a tunnel.
- Warns (but does not error, for now) on version mismatch

Mismatch logging will happen when initially connecting a document to an extension frame. Both the host and the guest will log version mismatches.

## Related Issue

TODO @fe-lix- may have opened this in JIRA, we've certainly talked about it

## Motivation and Context

Because host apps and guest extensions are almost always two different code bases, they can easily end up using different versions of the SDK. Backwards compatibility may not always be possible, so it's necessary to warn the developer or maintainer when this happens.

## How Has This Been Tested?

Tested by differential builds (building the host and guest SDKs at different versions locally) and then checking the logs in a running example site.

### Testing procedure

Check out this branch and do the following (all commands at repo root)

1. `npm run clean` to remove old build products
2. `npm run demo` to build examples and show current behavior
3. Visit the "Number Discussion" example
4. Open the Console and see _no_ version warnings
5. Ctrl-C to exit demo server
6. `npm pkg set version='99.99.99'` to temporarily set a new version string
7. `npm run build` to build libs with new version string
8. `node scripts/multi-server.mjs production` to run examples _without_ rebuilding them
9. Visit the "Number Discussion" example
10. Open the Console and see version warnings. Examples that have no build process and source the bundles directly will receive the changes, while examples that have build processes will not. This should show detected version mismatches in the console.
11. Ctrl-C to exit demo server
12. `git checkout -- package.json` to revert temporary version change

## Screenshots (if appropriate):
Screenshot 2023-01-04 at 5 21 12 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/uix-core/src/tunnel/tunnel-messenger.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+type Handshake = HandshakeAcceptedTicket | HandshakeOfferedTicket;
+type HandshakeAccepted = WrappedMessage<HandshakeAcceptedTicket>;
+type HandshakeOffered = WrappedMessage<HandshakeOfferedTicket>;
+type HandshakeMessage = HandshakeAccepted | HandshakeOffered;
+
+export class TunnelMessenger {
+  private myOrigin: string;
+  private remoteOrigin: string;
+  private logger: Console;
+  private versionWarnings = new Set<string>();
```
Comment (vkublytskyi) [inline packages/uix-core/src/tunnel/tunnel-messenger.ts:15]: Do we really need a state to manage warnings?

File: packages/uix-core/src/tunnel/tunnel-messenger.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    if (
+      !isPlainObject(tunnelData) ||
+      typeof tunnelData.version !== "string" ||
+      !(Reflect.has(tunnelData, "accepts") || Reflect.has(tunnelData, "offers"))
+    ) {
+      this.logMalformed(message);
+      return false;
+    }
+    const { version } = tunnelData;
+    if (version !== VERSION && !this.versionWarnings.has(version)) {
```
Comment (vkublytskyi) [inline packages/uix-core/src/tunnel/tunnel-messenger.ts:74]: Should we always expect exact versions match or we can allow patch or minor version differences?

File: packages/uix-core/src/tunnel/tunnel-messenger.ts (hunk 1 above)
Comment (zetlen) [reply packages/uix-core/src/tunnel/tunnel-messenger.ts:15]: With multiple extensions running and connecting additional frames, we may get repetitive version mismatch warnings. The state here just ensures only one warning for each mismatch.

File: packages/uix-core/src/tunnel/tunnel-messenger.ts (hunk 2 above)
Comment (zetlen) [reply packages/uix-core/src/tunnel/tunnel-messenger.ts:74]: (following in-person discussion) The "safe" version of this feature would require adding the `semver` library, which is 88K. Maybe we could assume that production version strings will always be `/\d+\.\d+\.\d+/`.

However! While we're still in `v0.x.x`, we have no API guarantees, and we can defer that question and just warn for any difference for now.

**Review by kokoc — APPROVED** [review]

**Pull request merged**